frontend/controller/app/identity.js

```diff
@@ -119,15 +119,19 @@ sbp('okTurtles.events/on', LOGIN, async ({ identityContractID, encryptionParams,
 
       // NOTE: users could notice that they leave the group by someone
       // else when they log in
-      const currentState = sbp('state/vuex/state')
-      if (!currentState.currentGroupId) {
-        const gId = Object.keys(currentState.contracts)
-          .find(cID => currentState[identityContractID].groups[cID] && !currentState[identityContractID].groups[cID].hasLeft)
-
-        if (gId) {
-          sbp('gi.app/group/switch', gId)
+      sbp('chelonia/queueInvocation', identityContractID, () => {
+        const currentState = sbp('state/vuex/state')
+        if (!currentState.currentGroupId && currentState[identityContractID]?.groups) {
+          const gId = Object.keys(currentState.contracts)
+            .find(cID => currentState[identityContractID].groups[cID] && !currentState[identityContractID].groups[cID].hasLeft)
+
+          if (gId) {
+            sbp('gi.app/group/switch', gId)
+          }
         }
-      }
+      }).catch(e => {
+        console.error('[LOGIN] Error setting current group ID', e)
+      })
```

Comment:
To me it seems like this code is really cutting things close. I don't see how this code is guaranteed to run after the identity contract has been sync'd. It seems possible to me that after `sbp('okTurtles.eventQueue/queueEvent', EVENT_HANDLED` and all the awaits in between, that this could code get run before the calls to `await sbp('chelonia/contract/retain', identityContractID)` / `await sbp('chelonia/contract/sync', identityContractID)`... in which case this could break because it would do nothing when it's supposed to do stuff.

Reply:
I think you may be right here... This should run _after_ syncing most of the time, if syncing fails, for example because one is offline, this could end up doing nothing.

Now, maybe this isn't really much of a concern. `/switch` does this:

```
    sbp('okTurtles.events/emit', SWITCH_GROUP, { contractID: groupId, isNewlyCreated })
    sbp('state/vuex/commit', 'setCurrentGroupId', { contractID: groupId, isNewlyCreated
````

The `SWITCH_GROUP` event seems to be unused, and we have an event listener on `JOINED_GROUP` that does that this block here is trying to do, i.e., set the current group ID if none is set. So, it'd look like if syncing isn't complete, then the event related to joining (switching) would be emitted later as more contracts are synced.